src/test/java/net/minecraftforge/debug/world/TeleporterTest.java

```diff
@@ -0,0 +1,190 @@
+        BLUE_POI = registerPOI(event.getRegistry(), new PointOfInterestType(new ResourceLocation(MODID, "blue_poi").toString(), PointOfInterestType.getAllStates(BLUE_TELEPORTER), 0, 1));
+    }
+    
+    private PointOfInterestType registerPOI(IForgeRegistry<PointOfInterestType> registry, PointOfInterestType poi)
+    {
+        poi.setRegistryName(new ResourceLocation(poi.toString()));
+        registry.register(poi);
+        try
+        {
+            ObfuscationReflectionHelper.findMethod(PointOfInterestType.class, "func_221052_a", PointOfInterestType.class).invoke(null, poi);
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+        return poi;
+    }
+    
+    private static class TeleporterBlock extends Block
+    {
+        final Supplier<PointOfInterestType> poi;
+        final RegistryKey<World> destWorldKey;
+        
+        private TeleporterBlock(Supplier<PointOfInterestType> poi, RegistryKey<World> destWorldKey)
+        {
+            super(Properties.from(Blocks.STONE));
+            this.poi = poi;
+            this.destWorldKey = destWorldKey;
+        }
+        
+        @Override
+        public void onFallenUpon(World worldIn, BlockPos pos, Entity entityIn, float fallDistance)
+        {
+            if (!worldIn.isRemote)
+                entityIn.changeDimension(((ServerWorld) worldIn).getServer().getWorld(worldIn.func_234923_W_() == this.destWorldKey ? World.field_234918_g_ : this.destWorldKey), new TestTeleporter(this.poi.get(), this));
+        }
+    }
+    
+    private static class TestTeleporter implements ITeleporter
+    {
+        final PointOfInterestType poi;
+        final Block teleporterBlock;
+        
+        private TestTeleporter(PointOfInterestType poi, Block teleporterBlock)
+        {
+            this.poi = poi;
+            this.teleporterBlock = teleporterBlock;
+        }
+        
+        @Override
+        public Optional<Result> findPortal(ServerWorld fromWorld, ServerWorld toWorld, Entity entity)
+        {
+            PointOfInterestManager poiManager = toWorld.getPointOfInterestManager();
+            int scale = TeleporterHelper.getPortalSearchRadius(fromWorld, toWorld);
+            BlockPos scaledPos = TeleporterHelper.getScaledPos(fromWorld, toWorld, new BlockPos(entity.getPositionVec()));
+            poiManager.ensureLoadedAndValid(toWorld, scaledPos, scale);
+            Optional<PointOfInterest> optional = poiManager.getInSquare((poiType) -> {
+               return poiType == this.poi;
+            }, scaledPos, scale, PointOfInterestManager.Status.ANY).sorted(Comparator.<PointOfInterest>comparingDouble((poi) -> {
+               return poi.getPos().distanceSq(scaledPos);
+            }).thenComparingInt((poi) -> {
+               return poi.getPos().getY();
+            })).findFirst();
```

Comment:
There is no reason to be using extended lambdas (with the braces) here it can just be `poiType -> poiType == this.poi` for example. Also instead of using `sorted` and `findFirst()` you can just use `min` where the current `sorted` call is and remove the `findFirst()` call

Reply:
Didn't realize `min` was a thing. Nifty!